### PR TITLE
#19 important section logic

### PR DESCRIPTION
**Pull request by calster00**
develop

## GitHub Board

**Issue link**

[#19 Important: Provide the logic for important section](https://github.com/ita-social-projects/dokazovi-fe/issues/19)

**Story link**

[#11 Story](https://github.com/ita-social-projects/dokazovi-be/issues/11)

## Code reviewers
- [ ] @github_username

## Summary of issue

- [ ]  Summary of issue
- [ ]  Summary of issue

## Summary of change

- [ ]  Summary of change
- [ ]  Summary of change

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -5,12 +5,14 @@ import ROUTER_CONFIG from './navigation/router-config';
...
 
 import Header from './lib/components/Header/Header';
+import ImportantContainer from './modules/main/components/ImportantContainer';
 
 const App: React.FC  =() => {
   return (
     <Suspense fallback={<div>Загрузка...</div>}>
       <div className="App">
         <Header/>
+        <ImportantContainer />
```
Comment (SergeOlabin) [inline src/App.tsx:15]: should me on the MainView, since it's valid only for main page

File: src/modules/main/components/ImportantContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+const ImportantContainer: React.FC= () => {
+  const posts = useSelector(selectImportant);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    Promise.resolve(Array(8).fill(MOCK_DATA)).then((d) => {
+      dispatch(loadImportant(d));
+    })
+    .catch(console.log);
+  }, []);
```
Comment (SergeOlabin) [inline src/modules/main/components/ImportantContainer.tsx:21]: pls load the data using redux:
- data is not passed to the action
- data is fetched on the action call
- data is obtained on the component using the selector

File: src/modules/main/store/actions.ts [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 } from './actionTypes';
 
-export function loadImportant(payload: IImportantItem[]): MainActions {
+export function loadImportant(payload: IPost[]): MainActions {
   return {
     type: LoadData.LOAD_IMPORTANT,
-    value: payload,
+    payload,
   };
 }
```
Comment (SergeOlabin) [inline src/modules/main/store/actions.ts:14]: `payload` 
should be obtainer inside the `loadImportant` function - this is the place where you have to load the data

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/modules/main/components/ImportantContainer.tsx (hunk 2 above)
Comment (calster00) [reply src/modules/main/components/ImportantContainer.tsx]: moved data fetching into `loadImportant`. maybe we should extract data fetching into separate function and pass it in action creators, so there is less repetition?

File: src/App.tsx (hunk 1 above)
Comment (calster00) [reply src/App.tsx]: fixed

File: src/modules/main/store/actions.ts [hunk 4]
```diff
@@ -1,16 +1,32 @@
...
+  IImportantAction,
 } from './actionTypes';
 
-export function loadImportant(payload: IImportantItem[]): MainActions {
-  return {
-    type: LoadData.LOAD_IMPORTANT,
-    value: payload,
-  };
+export function loadImportant(): ThunkAction<
+  void, // return type
```
Comment (SergeOlabin) [inline src/modules/main/store/actions.ts:17]: `Promise<any>` ?

File: src/modules/main/components/ImportantContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  useEffect(() => {
+    dispatch(loadImportant());
+  }, [dispatch]);
+
+  return (
+    <div>
+      <Carousel>
+        {posts.map((p, i) => (
+          // eslint-disable-next-line react/no-array-index-key
```
Comment (SergeOlabin) [inline src/modules/main/components/ImportantContainer.tsx:22]: pls fix

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/modules/main/store/actions.ts (hunk 4 above)
Comment (calster00) [reply src/modules/main/store/actions.ts]: fixed

File: src/modules/main/components/ImportantContainer.tsx (hunk 2 above)
Comment (SergeOlabin) [reply src/modules/main/components/ImportantContainer.tsx]: might be a good idea though!
when we will use the real BE, we could create a service to fetch the data and use it in the action.

File: src/modules/main/components/ImportantContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+const selectImportant = (state: IRootState) => state.main.important;
+
+const ImportantContainer: React.FC = () => {
+  const posts = useSelector(selectImportant);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(loadImportant());
+  }, [dispatch]);
```
Comment (SergeOlabin) [inline src/modules/main/components/ImportantContainer.tsx:16]:
```suggestion
  }, []);
```

**Review by SergeOlabin — APPROVED** [review]

**Pull request merged**